### PR TITLE
feat(core): add rust mr_boxington tool option and use flag

### PR DESCRIPTION
**Pull request by jdx**
Rust users currently need to configure a Cargo command wrapper separately to use Mr Boxington. Add `mr_boxington = true` to the Rust tool entry and a scoped `mise use --tool-option KEY=VALUE` flag, making setup a single command:

```sh
mise use --tool-option mr_boxington=true rust mr-boxington
```

The Rust option supplies the existing Cargo wrapper automatically while keeping mbx independently versioned and locked. Explicit Cargo wrappers take precedence; missing mbx configuration produces an actionable error. Project, task, and explicit exec overrides are respected, and project wrapping is ignored in safe mode. Cargo must still resolve through mise.

`--tool-option` is repeatable and applies to the following tool, like `--postinstall`. It preserves existing options, accepts the normal option value types, rejects duplicate/inline conflicts, and selects a TOML config when necessary. Includes generated CLI documentation and a Rust setup guide.

Validation:
- `mise run lint-fix`
- `cargo clippy --workspace --all-features --all-targets -- -D warnings`
- 28 tool-option unit tests
- E2E suites for Rust/mbx integration, scoped tool options, existing postinstall behavior, command wrappers, and exec recursion with and without shims
- Generated usage/completions; CLI reference tests pass

*AI-assisted — Tool: Codex; model: openai/unavailable; version: unavailable.*

---

> [!NOTE]
> **Medium Risk**
> Changes how Cargo is resolved on PATH and in shims for Rust projects, with new runtime wrapper logic in exec/tasks; mistakes could break builds or bypass intended wrappers, though explicit wrappers and safe mode limits are preserved.
> 
> **Overview**
> Adds **`mise use --tool-option KEY=VALUE`** (repeatable, scoped to the next tool like `--postinstall`) so tool entries can be written with inline options from the CLI. Values follow normal TOML option typing; duplicates, inline conflicts, and non-TOML targets are rejected, and option-bearing `use` calls steer writes to a default `mise.toml` when needed.
> 
> Rust gains **`mr_boxington = true`**, which auto-installs the existing **Cargo → mbx** command wrapper when `mr-boxington` is also in `[tools]`, without a manual `[wrappers.cargo]` block. Wrapper selection now depends on the **resolved toolset** (project/task/exec overrides, safe mode for project opt-ins, explicit wrappers still win). **PATH** and shim generation were adjusted so tool-derived wrappers apply after env resolution; **exec** and **tasks** ensure wrapper shims exist at runtime and preserve explicit exec tool overrides in the shim dispatch env. Env caching includes tool options so wrapper changes invalidate correctly.
> 
> Docs and generated CLI help cover the new flag and the updated Mr Boxington setup flow; e2e tests cover `--tool-option` validation and mbx routing across exec, tasks, shims, and edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 3e04c5c96e67e1024ffc6c6f0d31662539e74cba. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

- **New Features**
  - Added repeatable `--tool-option KEY=VALUE` support to `mise use` for per-tool configuration.
  - Rust Cargo commands can route through Mr Boxington during executions, tasks, activated shells, and shims.
  - Existing Cargo wrappers and project-level settings take precedence where applicable.

- **Bug Fixes**
  - Improved consistency of tool options, explicit tool selections, command wrappers, and environments.

- **Documentation**
  - Updated CLI and Rust documentation with setup requirements, behavior details, and examples.

- **Tests**
  - Added coverage for validation, overrides, dry runs, missing tools, and error handling.

### CODE REVIEW

File: src/cli/use.rs [hunk 1]
```diff
@@ -246,7 +302,7 @@ impl Use {
                             resolve_options.use_locked_version = false;
                         }
                         let mut tvr = tvr;
-                        if target.postinstall.is_some() {
+                        if target.has_options() {
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/cli/use.rs:305]: Verified against the current code: `install_all_versions` calls `init_request_options`, which reapplies matching configured options before installation and persistence. The reported loss did not reproduce. Commit 621d4db91 adds explicit regression assertions for preserving Rust components and existing inline options during a versionless `mise use --tool-option` update; both pass.

*AI-assisted — Tool: Codex; model: openai/unavailable; version: unavailable.*

File: src/config/mod.rs [hunk 2]
```diff
@@ -1428,7 +1436,12 @@ impl Config {
             },
         )
         .await?;
-        if !load_command_wrappers(&self.config_files)?.is_empty() {
+        if !load_command_wrappers(
+            &self.config_files,
+            self.get_tool_request_set().await?.tools.values().flatten(),
+        )?
+        .is_empty()
+        {
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/config/mod.rs]: Fixed in 621d4db91. Task environment resolution now uses the effective task tool requests, and task/exec dispatch creates missing command-wrapper shims without removing wrappers needed by other tasks. Regression tests start with project wrapping disabled and no wrapper shim, then verify task-level and explicit exec opt-ins, as well as an unwrapped sibling task. The targeted E2E suites, lint-fix, and strict Clippy checks pass.

*AI-assisted — Tool: Codex; model: openai/unavailable; version: unavailable.*

**Comment by jdx** [comment]
Addressed the safe-mode documentation feedback in 621d4db91: the Rust integration section now states that safe mode ignores opt-ins from project-scoped Rust entries.

*AI-assisted — Tool: Codex; model: openai/unavailable; version: unavailable.*

File: src/task/task_executor.rs [hunk 3]
```diff
@@ -2095,6 +2095,22 @@ impl TaskExecutor {
...
+                .map(|(backend, versions)| {
+                    (
+                        backend.clone(),
+                        versions.requests.clone(),
+                        versions.source.clone(),
+                    )
+                })
+                .collect(),
+        );
+        crate::shims::ensure_command_wrapper_shims(config, &toolset)?;
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/task/task_executor.rs:2098]: Fixed in 3e04c5c96. The environment cache key now includes each resolved tool request’s options alongside its version, so wrapped and unwrapped tasks cannot share a cached PATH. The Rust E2E suite now enables MISE_ENV_CACHE and alternates task opt-ins/opt-outs. Wrapper path selection also moved to the resolved toolset, eliminating premature tool-template resolution during sourced environment loading. All 3,579 unit tests, targeted wrapper E2E suites, lint-fix, and strict Clippy pass.

*AI-assisted — Tool: Codex; model: openai/unavailable; version: unavailable.*
